### ISSUE TITLE
Recognize the @implements, @record and @interface annotations

### ISSUE THREAD

**Issue by ChadKillingsworth** [Polymer/polymer-analyzer#595]
When using mixins with Closure compiler, you frequently end up writing code like this:

```js
/** @record */
class MyMixinInterface {}

/** @param {function(new:Polymer.Element)} */
function myMixin(Superclass) {
  class MixedElement extends Superclass {}
  return MixedElement;
}

/**
 * @constructor
 * @extends {Polymer.Element}
 * @implements {MyMixinInterface}
 */
const MixedElement = myMixin(Polymer.Element);

class MyElement extends MixedElement {}
```
In an ideal world, the analyzer would recognize the existing annotations. Currently there are a few problems:

 1. The alias created for the `MixedElement` definition isn't traced through. While there is no end to an alias lookup cycle in general, a single hop lookup should be doable.
 2. The analyzer doesn't recognize that `@implements` is essentially the same as `@mixes`.
 3. The analyzer doesn't recognize the `@record` to be essentially the same as `@polymerMixin`.

There's enough hoops to jump through with Closure-compiler already. If the analyzer could infer from what is already written, it would make closure-compiler developers a lot happier.

**Comment by rictic** [comment]
I don't think that `@implements` or `@record` are the same as `@mixes` or `@polymerMixin`. Even if it's the case that `@implements` and `@record` mean the same thing in an ES6 Class Mixin context, they are also used in other contexts and we need some reliable signal in order to determine when a mixin is being declared `@polymerMixin` and when one is being mixed in (rather than an its interface is satisfied in some other way).

I think ideally we will alleviate this by doing more inference, so that the annotations are necessary less often.

**Comment by justinfagnani** [comment]
We might be able to recognize these, but I think we'd need some additional annotations still. An interface is simply a different concept from a mixin, in that the mixin actually provides an implementation. We could support Closure interfaces and add those to the docs to solve it, but we also want to link to the actual mixin definition itself so that a user can see how to apply it, so I think we'd need some sort of annotation on the `@record` that links it to the mixin function.

**Comment by ChadKillingsworth** [comment]
What I want out of this isn't documentation - I want the linter to be able to recognize where a property exists (or actually doesn't exit). And the same thing for polymer-rename.
